conda/gateways/repodata/__init__.py

```diff
@@ -579,7 +579,9 @@ def load_state(self):
         """
         try:
             self.load(state_only=True)
```

Comment:
I notice we also load the state in the `load()` method. It happens that we always load the state in this method first, to avoid parsing `repodata.json` twice when it is outdated. Conceivably we would need the same checks in load() however.